### PR TITLE
feat: initialize `campaign` module

### PR DESCRIPTION
**Pull request by lubtd**
This initializes the `campaign` module and implement some utilities and refactos for future PRs like `campaign` test keeper and removing IBC placeholders

For `launch` and `campaign` bidirectional relation, we keep an instance of the other module keeper in each keeper. This is in fact not a big concern since keeper instance is stateless, each keeper methods get a `Context` object parameter

### CODE REVIEW

File: x/campaign/keeper/keeper.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	cdc codec.Marshaler,
+	storeKey,
+	memKey sdk.StoreKey,
+	launchKeeper types.LaunchKeeper, bankKeeper types.BankKeeper, profileKeeper types.ProfileKeeper,
+) *Keeper {
+	return &Keeper{
+		cdc:      cdc,
+		storeKey: storeKey,
+		memKey:   memKey,
+		launchKeeper: launchKeeper, bankKeeper: bankKeeper, profileKeeper: profileKeeper,
```
Comment (Pantani) [inline x/campaign/keeper/keeper.go:35]: needs to run the gofmt

File: x/campaign/types/codec.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	// this line is used by starport scaffolding # 3
+
+	msgservice.RegisterMsgServiceDesc(registry, &_Msg_serviceDesc)
+}
+
+var (
+	amino     = codec.NewLegacyAmino()
```
Comment (Pantani) [inline x/campaign/types/codec.go:21]:
```suggestion
	Amino     = codec.NewLegacyAmino()
```

File: x/campaign/types/genesis_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package types_test
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/spn/x/campaign/types"
+	"testing"
```
Comment (Pantani) [inline x/campaign/types/genesis_test.go:6]:
```suggestion
	"testing"

	"github.com/stretchr/testify/require"
	"github.com/tendermint/spn/x/campaign/types"
```

File: x/campaign/types/genesis_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		{
+			desc:     "valid genesis state",
+			genState: &types.GenesisState{
+				// this line is used by starport scaffolding # types/genesis/validField
+			},
+			valid: true,
+		},
+		// this line is used by starport scaffolding # types/genesis/testcase
+	} {
+		tc := tc
```
Comment (Pantani) [inline x/campaign/types/genesis_test.go:29]:
```suggestion
```

**Review by ilgooz — APPROVED** [review]

**Review by fadeev — APPROVED** [review]
👍

**Review by Pantani — APPROVED** [review]

**Pull request merged**